akd/planner/registry.py

```diff
@@ -0,0 +1,286 @@
+"""
+Agent Registry for the AKD Planner module.
+
+This module provides agent registration and discovery capabilities for the AKD framework.
+"""
+
+import importlib
+import json
+import os
+from typing import Any, Dict, List, Optional, Tuple, Type, Union
+from datetime import datetime, timezone
+
+from loguru import logger
+from pydantic import BaseModel, Field
+
+from akd._base import IOSchema
+from .config import AgentRegistryConfig
+
+
+class FieldDefinition(BaseModel):
+    """Individual field definition for agent schemas."""
+    
+    name: str = Field(description="Field name")
+    type: str = Field(description="Field type")
```

Comment:
Are these required fields. Then we could use elipses `...` in the `Field(..., descriptoin="...")`